pkg/odo/cli/catalog/describe/service.go

```diff
@@ -42,20 +62,88 @@ func NewDescribeServiceOptions() *DescribeServiceOptions {
 
 // Complete completes DescribeServiceOptions after they've been created
 func (o *DescribeServiceOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	isSVCSupported, err := svc.IsCSVSupported()
+	if err != nil {
+		isSVCSupported = false
+	}
+	if strings.Contains(args[0], "/") && isSVCSupported {
+		tmpOptrList := strings.Split(args[0], "/")
+		o.ServiceType = tmpOptrList[0]
+		o.CustomResource = tmpOptrList[1]
+		o.isOperator = true
+	} else {
+		o.serviceName = args[0]
+	}
+
 	o.Context, err = genericclioptions.NewContext(cmd, true)
-	o.serviceName = args[0]
 
 	return
 }
 
 // Validate validates the DescribeServiceOptions based on completed values
 func (o *DescribeServiceOptions) Validate() (err error) {
+
+	if o.isOperator {
+
+		if o.ServiceType == "" || o.CustomResource == "" {
+			return fmt.Errorf("invalid service name, use the format <operator-type>/<crd-name>")
+		}
+		// make sure that CSV of the specified ServiceType exists
+		o.CSV, err = o.KClient.GetClusterServiceVersion(o.ServiceType)
+		if err != nil {
+			// error only occurs when OperatorHub is not installed.
+			// k8s does't have it installed by default but OCP does
+			return err
+		}
+
+		// Get the specific CR that matches "kind"
+		crs := o.KClient.GetCustomResourcesFromCSV(&o.CSV)
+
+		var cr *olm.CRDDescription
+		hasCR := false
+		for _, custRes := range *crs {
+			c := custRes
+			if c.Kind == o.CustomResource {
+				cr = &c
+				hasCR = true
+				break
+			}
+		}
+		if !hasCR {
+			return fmt.Errorf("the %s resource doesn't exist in specified %s operator", o.CustomResource, o.ServiceType)
+		}
+
+		o.CR = cr
+		return nil
+	}
 	o.service, o.plans, err = svc.GetServiceClassAndPlans(o.Client, o.serviceName)
 	return err
+
 }
 
 // Run contains the logic for the command associated with DescribeServiceOptions
 func (o *DescribeServiceOptions) Run() (err error) {
+	if o.isOperator {
+		return o.operatorRun()
+	}
+	return o.serviceCatalogRun()
+
+}
+func (o *DescribeServiceOptions) operatorRun() (err error) {
+	if log.IsJSON() {
+		machineoutput.OutputSuccess(o.CR)
+		return
+	}
+	output, err := yaml.Marshal(svc.ConvertCRDToRepr(o.CR))
+	if err != nil {
+		return err
+	}
+
+	fmt.Print(string(output))
```

Comment:
Is this left by mistake? Asking because I don't think we use `fmt` anywhere to print to stdout in odo code. I could be wrong.

Reply:
It is being used at multiple places https://github.com/openshift/odo/search?q=fmt.Print